lib/vault.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const { exec } = require('child_process');
+const fs = require('fs');
+
+const service = process.env.SERVICE_SYSTEM_CODE;
+const region = (process.env.REGION || 'LOCAL').toLowerCase();
+const vaultPath = `secret/teams/origami/${service}/${region}`;
+const envFile = '.env';
+
+if (!service) {
+    console.warn(
+        'Could not load environment variables from Vault. Service name is undefined.',
+        'Set the environment variable "SERVICE_SYSTEM_CODE" in your Makefile.'
+    );
+    process.exit(1);
+}
+
+// Update an existing .env file with environment variables from Vault.
+exec(`vault read -format=json ${vaultPath}`, (err, stdout, stderr) => {
+    const result = err ? null : JSON.parse(stdout);
+
+    // Handle errors.
+    if (err || !result.data) {
+        console.warn(
+            `Could not load environment variables from Vault (${vaultPath}).`,
+            (stdout ? `\nstdout: ${stdout}` : ''),
+            (stderr ? `\nstderr: ${stderr}` : '')
+        );
+        process.exit(1);
```

Comment:
We should avoid using `process.exit(1)` and use `process.exitCode = 1;` instead.

Calling `process.exit(1)` forces the process to exit as quickly as possible, even if there are still asynchronous operations pending that have not yet completed fully. E.G. The `console.warn` on the line above may not be written to `process.stdout` because `console.warn` may sometimes be asynchronous.

Reply:
TIL! Thanks. I've replaced the with throwing an error.